inference-engine/tests/functional/plugin/cpu/single_layer_tests/topk.cpp

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "functional_test_utils/ov_tensor_utils.hpp"
+#include "common_test_utils/data_utils.hpp"
+#include "test_utils/cpu_test_utils.hpp"
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "ngraph_functions/builders.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+using namespace ov::test;
+
+namespace CPULayerTestsDefinitions {
+using topKParams = std::tuple<
+    ngraph::element::Type, // data precision
+    InputShape, // data shape
+    std::int64_t, // axis
+    std::vector<std::int64_t>, // k value for each infer request
+    ngraph::opset4::TopK::Mode,
+    ngraph::opset4::TopK::SortType>;
+
+class TopKLayerCPUTest : public testing::WithParamInterface<topKParams>, public ov::test::SubgraphBaseTest, public CPUTestsBase {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<topKParams> obj) {
+        ngraph::element::Type inputPrecision;
+        InputShape inputShape;
+        std::int64_t axis;
+        std::vector<std::int64_t> kVals;
+        ngraph::opset4::TopK::Mode mode;
+        ngraph::opset4::TopK::SortType sortType;
+        std::tie(inputPrecision, inputShape, axis, kVals, mode, sortType) = obj.param;
+
+        std::ostringstream result;
+        result << inputPrecision << "_" << "IS=" << CommonTestUtils::partialShape2str({ inputShape.first }) << "_" << "TS=(";
+        for (const auto& shape : inputShape.second) {
+            result << CommonTestUtils::vec2str(shape) << "_";
+        }
+
+        result << ")_axis=" << axis << "_k_vals=" << CommonTestUtils::vec2str(kVals) << "_mode=" << mode << "_sortType=" << sortType;
+        return result.str();
+    }
+
+protected:
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+
+        ov::runtime::Tensor data_tensor;
+        const auto& dataPrecision = funcInputs[0].get_element_type();
+        const auto& dataShape = targetInputStaticShapes.front();
+        if (funcInputs[0].get_element_type().is_real()) {
+            data_tensor = ov::test::utils::create_and_fill_tensor(dataPrecision, dataShape, 10, 0, 1000);
+        } else {
+            data_tensor = ov::test::utils::create_and_fill_tensor(dataPrecision, dataShape);
+        }
+
+        const std::vector<std::int64_t> kValues = std::get<3>(this->GetParam());
+        const auto& kPrecision = funcInputs[1].get_element_type();
+        const auto& kShape = targetInputStaticShapes[1];
+        auto kTensor = ov::runtime::Tensor{ kPrecision, kShape };
+        auto data = static_cast<std::int32_t*>(kTensor.data());
+
+        if (inferRequestNum < kValues.size()) {
+            data[0] = static_cast<std::int32_t>(kValues[inferRequestNum++]);
+        } else {
+            IE_THROW() << "k for current infer request wasn't defined.";
+        }
```

Comment:
Can we generate k value randomly? Just using this counter as a seed for the random number generator (to obtain different k value) and using the shape to define its range. Or we want control the k values manually by some reason?

Reply:
Why not? Random k value generation has been added.